src/aap_eda/core/utils/decision_environment.py

```diff
@@ -0,0 +1,39 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from aap_eda.core.enums import ImagePullPolicy
+
+
+def convert_pull_policy_to_frontend(pull_policy: str) -> str:
+    """Convert internal pull policy enum to frontend-friendly string."""
+    policy_mapping = {
+        ImagePullPolicy.ALWAYS: "always",
+        ImagePullPolicy.NEVER: "never",
+        ImagePullPolicy.IF_NOT_PRESENT: "missing",
+    }
+    return policy_mapping.get(pull_policy, "always")
```

Comment:
Even if you want to continue with the current approach of these utils, this line is a potential bug that masks the default policy.